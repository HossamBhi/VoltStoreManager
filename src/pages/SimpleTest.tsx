import React, {useEffect, useState} from 'react';
import {Pressable, Text, View} from 'react-native';

const SimpleTest = () => {
  const [text, setText] = useState('');
  const styles = {backgroundColor: 'red'};

  useEffect(() => {
    console.log('It is screen');
  }, []);
  return (
    <View style={styles}>
      <Text testID="textHello">{text}</Text>
      <Pressable testID="helloButton" onPress={() => setText('Hello Test')}>
        <Text>Press me</Text>
      </Pressable>
    </View>
  );
};

export default SimpleTest;
