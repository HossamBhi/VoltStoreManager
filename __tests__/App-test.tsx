/**
 * @format
 */

import 'react-native';
import React from 'react';
// import App from '../App';

// Note: test renderer must be required after react-native.
import {create, act} from 'react-test-renderer';
import SimpleTest from '../src/pages/SimpleTest';

// it('renders correctly', () => {
//   renderer.create(<App />);
// });

// it('Menu page', () => {
//   const menuPage = renderer.create(<Menu />);
//   expect(menuPage).toMatchSnapshot();
// });
const simplePage = create(<SimpleTest />);

it.skip('Simple page', () => {
  expect(simplePage).toMatchSnapshot();
});

test.skip('Test Button', () => {
  const helloBtn = simplePage.root.findByProps({testID: 'helloButton'}).props;
  act(() => helloBtn.onPress());

  const helloText = simplePage.root.findByProps({testID: 'textHello'}).props;
  expect(helloText.children).toEqual('Hello Test');
});
