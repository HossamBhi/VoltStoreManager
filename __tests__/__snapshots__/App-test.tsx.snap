// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <Text
    testID="textHello"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="helloButton"
  >
    <Text>
      Press me
    </Text>
  </View>
</View>
`;
